drivers/adc/adc_shell.c

```diff
@@ -0,0 +1,447 @@
+	{ .string = "ADC_GAIN_32", .gain = ADC_GAIN_32 },
+	{ .string = "ADC_GAIN_64", .gain = ADC_GAIN_64 }
+};
+
+static int cmd_adc_gain(const struct shell *shell, size_t argc, char **argv)
+{
+	int retval = -EINVAL;
+	struct device *adc_dev;
+	int chosen_adc;
+	int i;
+
+	chosen_adc = get_adc_from_list(argv[args_indx.parent_adc]);
+	if (chosen_adc < 0) {
+		return -EINVAL;
+	}
+
+	adc_dev = device_get_binding(adc_list[chosen_adc].device_name);
+	if (adc_dev == NULL) {
+		shell_error(shell, "ADC device not found");
+		return -ENODEV;
+	}
+	for (i = 0; i < ARRAY_SIZE(gain_list); i++) {
+		if (!strcmp(argv[0], gain_list[i].string)) {
+			adc_list[chosen_adc].channel_config.gain =
+				gain_list[i].gain;
+			retval = adc_channel_setup(adc_dev,
+					&adc_list[chosen_adc].channel_config);
+			LOG_DBG("Channel setup returned %i\n", retval);
+			break;
+		}
+	}
+	return retval;
+}
+
+
+static int cmd_adc_acq(const struct shell *shell, size_t argc, char **argv)
+{
+	int retval = 0;
+	struct device *adc_dev;
+	int chosen_adc;
+	u16_t acq_time;
+
+	if (argc != args_no.acq_time) {
+		shell_fprintf(shell, SHELL_NORMAL,
+				"Usage: acq_time <time> <unit>\nunits: us, ns, ticks\n");
+		return 0;
+	}
+
+	chosen_adc = get_adc_from_list(argv[args_indx.adc]);
+	if (chosen_adc < 0) {
+		return -EINVAL;
+	}
+
+	adc_dev = device_get_binding(adc_list[chosen_adc].device_name);
+	if (adc_dev == NULL) {
+		shell_error(shell, "ADC device not found");
+		return -ENODEV;
+	}
+
+	if (!isdigit(argv[args_indx.conf][0])) {
+		return -EINVAL;
+	}
+	acq_time = (u16_t)strtol(argv[args_indx.conf], NULL, 10);
+	if (!strcmp(argv[args_indx.acq_unit], "us")) {
+		adc_list[chosen_adc].channel_config.acquisition_time =
+			ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, acq_time);
+	} else if (!strcmp(argv[args_indx.acq_unit], "ns")) {
+		adc_list[chosen_adc].channel_config.acquisition_time =
+			ADC_ACQ_TIME(ADC_ACQ_TIME_NANOSECONDS, acq_time);
+	} else if (!strcmp(argv[args_indx.acq_unit], "ticks")) {
+		adc_list[chosen_adc].channel_config.acquisition_time =
+			ADC_ACQ_TIME(ADC_ACQ_TIME_TICKS, acq_time);
+	} else {
+		adc_list[chosen_adc].channel_config.acquisition_time =
+			ADC_ACQ_TIME_DEFAULT;
+	}
+	retval = adc_channel_setup(adc_dev,
+			&adc_list[chosen_adc].channel_config);
+	LOG_DBG("Channel setup returned %i\n", retval);
+	return retval;
+}
+static int cmd_adc_reso(const struct shell *shell, size_t argc, char **argv)
+{
+	int retval = 0;
+	struct device *adc_dev;
+	int chosen_adc;
+
+	if (argc != args_no.resolution ||
+			!isdigit((unsigned char)argv[args_indx.conf][0])) {
+		shell_fprintf(shell, SHELL_NORMAL,
+				"Usage: resolution <resolution>\n");
+		return 0;
+	}
+
+	chosen_adc = get_adc_from_list(argv[args_indx.adc]);
+	if (chosen_adc < 0) {
+		return -EINVAL;
+	}
+
+	adc_dev = device_get_binding(adc_list[chosen_adc].device_name);
+	if (adc_dev == NULL) {
+		shell_error(shell, "ADC device not found");
+		return -ENODEV;
+	}
+	if (!isdigit(argv[args_indx.conf][0])) {
```

Comment:
newline after if statements

Reply:
@sknsean Newline should be after the opening brace, just as it is. What do you mean?

Reply:
newline after closing bracket of if statement, for readability

Reply:
I think you're nitpicking ;-)

Reply:
not at all :-D